q2_phylogenomics/_assemble.py

```diff
@@ -0,0 +1,121 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import tempfile
+import skbio
+import pandas as pd
+import os.path
+import hashlib
+
+from q2_types.per_sample_sequences import (
+    SingleLanePerSamplePairedEndFastqDirFmt)
+from q2_types.bowtie2 import Bowtie2IndexDirFmt
+from q2_types.feature_data import (DNASequencesDirectoryFormat,
+                                   DNAFASTAFormat)
+
+from ._util import run_command
+from ._format import (
+        BAMFilesDirFmt, PileUpFilesDirFmt,
+        BAMFormat, PileUpTSVFormat)
+
+
+def map_paired_reads(demux: SingleLanePerSamplePairedEndFastqDirFmt,
+                     reference: Bowtie2IndexDirFmt,
+                     mismatches_per_seed: int = 1,
+                     ceil_func: str = '0,0.3',
```

Comment:
the bowtie2 --n-ceil option looks really flexible, more than we probably need here. Looks like you are already always using a linear function, so maybe also just use a linear intercept? In which case you can change this to a `float` and Q2 can handle the validation.